[5.0] Check for a file to provide parameter value in Dispatcher

mattmcdev opened the pull request:
If we have a files object in source, and the value would otherwise be null, and we have the file we’re referencing, then return that instead.

This fixes #8137

taiyaei commented on src/Illuminate/Bus/Dispatcher.php:
```diff
@@ -154,6 +154,10 @@ protected function getParameterValueForCommand($command, ArrayAccess $source,
+			if (gettype($source->files) === 'object' && is_null($source[$parameter->name]) && $source->file($parameter->name))
+            {
+              return $source->file($parameter->name);
+            }
```
L158-160, tabs for CS

taiyaei commented on src/Illuminate/Bus/Dispatcher.php:
```diff
@@ -154,6 +154,10 @@ protected function getParameterValueForCommand($command, ArrayAccess $source,
+            {
+              return $source->file($parameter->name);
+            }
 			return $source[$parameter->name];
```
Newline before return

GrahamCampbell commented:
Please squash to one commit and revert the change to the file permissions.

mattmcdonald-uk commented:
@GrahamCampbell Not sure what you mean by changed file permissions (as in, I wasn't aware I'd changed any file permissions!)

GrahamCampbell commented:
They're still wrong. Look:

![image](https://cloud.githubusercontent.com/assets/2829600/6828775/671122cc-d307-11e4-9bee-4f8916218596.png)

Commit:
Check for a file to provide parameter value

If we have a files object in source, and the value would otherwise be null, and we have the file we’re referencing, then return that instead

mattmcdonald-uk commented:
@GrahamCampbell Sorted!

GrahamCampbell commented:
Thanks.

taylorotwell commented:
see me comment on issue: https://github.com/laravel/framework/issues/8137
